src/core/core.js

```diff
@@ -258,8 +259,12 @@ export default class Core {
           const mergedResults = this.storage.get(StorageKeys.VERTICAL_RESULTS)
             .append(data[StorageKeys.VERTICAL_RESULTS]);
           this.storage.set(StorageKeys.VERTICAL_RESULTS, mergedResults);
+          if (data[StorageKeys.DIRECT_ANSWER].answer) {
```

Comment:
In this instance we're guarding the setting of `StorageKeys.DIRECT_ANSWER`, but below we just set it without checking `data[StorageKeys.DIRECT_ANSWER].answer`. Do we need to check this value in both cases? If so, we could move that logic outside of the `if(query.append)` conditional and only have it once.

Reply:
I don't think we need to check this value in both cases. The first result set from this vertical search may or may not have direct answers from live api, which is ok so I didn't place a check there. But with param query.append allowing for search result of new query to add on to the existing results, if this new search result contains a new directAnswer, then we would override it (since I believe there should only be just 1 directAnswer return). Otherwise, if this new search result doesn't have a directAnswer, then we keep the previous directAnswer from current result set.